File: Code/Components/DanpheEMR.ServerModel/AdmissionModels/BedModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class BedModel
    {
        [Key]
        public int BedId { get; set; }
        public string BedCode { get; set; }
        public int BedNumber { get; set; }
        public int WardId { get; set; }
        public bool IsOccupied { get; set; }
        public int? CreatedBy { get; set; }
        public DateTime? CreatedOn { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public DateTime? HoldedOn { get; set; }
        public bool IsActive { get; set; }
        public WardModel Ward { get; set; }
        public bool? IsReserved { get; set; }
        public bool? OnHold { get; set; }
        [NotMapped]
        public int BedNumFrm { get; set; }
        [NotMapped]
        public int BedNumTo { get; set; }
        //[NotMapped]
        //public string ReservedByPatient { get; set; }
        //[NotMapped]
        //public DateTime? ReservedForDate { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/adt/shared/bed.model.ts
/// <reference path="ward.model.ts" />
//import { Ward } './ward.model';
//import { BedType } './bedtype.model';
import {
  FormBuilder,
  FormGroup,
  Validators
} from '@angular/forms';
import * as moment from 'moment';
import { ENUM_DateTimeFormat } from '../../shared/shared-enums';
import { BedFeature } from './bedfeature.model';
export class Bed {
  public BedId: number = 0;
  public BedCode: string = null;
  public BedFeature: BedFeature = new BedFeature();
  public BedNumber: string = null;
  // public BedNumFrm: string = null;
  // public BedNumTo: string = null;
  public WardId: number = null;
  public IsOccupied: boolean = false;
  public IsReserved: boolean = false;

  public CreatedBy: number = 0;
  public ModifiedBy: number = null;
  public IsActive: boolean = true;

  public CreatedOn: string = null;
  public ModifiedOn: string = null;

  public ReservedByPatient: string = null;
  public ReservedForDate: string = null;


  public BedMainValidator: FormGroup = null;
  //public Ward: Ward = new Ward();
  //public BedType: BedType = new BedType();

  constructor() {
    this.CreatedOn = moment().format(ENUM_DateTimeFormat.Year_Month_Day_Hour_Minute);
    var _formbuilder = new FormBuilder();
    this.BedMainValidator = _formbuilder.group({
      'WardId': ['', Validators.compose([Validators.required])],
      'BedNumber': ['', Validators.compose([Validators.required])]
    });
  }
  public IsDirty(fieldName): boolean {
    if (fieldName == undefined)
      return this.BedMainValidator.dirty;
    else
      return this.BedMainValidator.controls[fieldName].dirty;
  }

  public IsValid(): boolean { if (this.BedMainValidator.valid) { return true; } else { return false; } } public IsValidCheck(fieldName, validator): boolean {
    if (fieldName == undefined) {
      return this.BedMainValidator.valid;

    }

    else
      return !(this.BedMainValidator.hasError(validator, fieldName));
  }
}
